databuilder/tables/beta/tpp.py

```diff
@@ -58,6 +59,23 @@ class practice_registrations(EventFrame):
         ],
     )
 
+    def active_for_patient_at(self, date):
```

Comment:
A clear improvement.

Reply:
> I'm not quite sure about the term "active" for practice registrations and addresses, but couldn't immediately think of a better one. "Primary" maybe? It would be good to have a term we consistently use here.

at first I thought "current" instead of "active" would be better, but then I realised that "current" is misleading if we look up an address or registration in the past. the only idea I had was to remove the prefix altogether and mention the table/data it refers to but then the functions have different names, e.g., 
- `practice_registrations.registration_for_patient_at()`
- `addresses.address_for_patient_at()`

Reply:
I think we need something that conveys that a patient might have more than one of these things (address or practice registration) at a given time and we're somehow selecting the "best" one, which `address_for_patient_at()` doesn't quite do. I'm still not sold on "active" but I can't think of anything better.

Reply:
or `pick_for_patient_at`

Reply:
@inglesp and I agreed that "active" (and related terms like "primary") are confusing as they suggest this is somehow an official status recorded in the EHR, rather than just our best guess at picking something.

We've therefore gone for the minimalist `for_patient_on()` on the basis that the full explanation lives in the docs and the less we imply in the method name, the better.